src/ui/public/dialogues/confirm_dialogue.js

```diff
@@ -0,0 +1,63 @@
+import uiModules from 'ui/modules';
+import confirmDialogueTemplate from './confirm_dialogue.html';
+
+const module = uiModules.get('kibana');
+
+module.factory('showConfirmDialogue', function (ModalDialogue, $q, $compile, $rootScope) {
+  /**
+   * Shows a modal confirmation dialog with the given message.
+   *
+   * @param {String} message
+   * @param {String} yesButtonText - Defaults to Okay, supply a custom string to override.
+   * @param {String} noButtonText - Defaults to Cancel, supply a custom string to override.
+   * @return {Promise<boolean>} Returns a promise that will be resolved to true if the user
+   * clicks the yes/okay button and rejected if the user clicks the no/cancel button.
+   */
+  return function showConfirmDialogue(message, yesButtonText = 'Okay', noButtonText = 'Cancel') {
+    const deferred = $q.defer();
+
+    const dialogueScope = $rootScope.$new();
+
+    let modalDialogue = undefined;
+
+    dialogueScope.onYes = () => {
+      dialogueScope.$destroy();
+      modalDialogue.destroy();
+      deferred.resolve(true);
```

Comment:
Do we still need to return `true` and `false` when we resolve and reject? I think we can remove them since they don't seem to serve a purpose. This will allow us to simplify the tests a bit.